ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionComponentInstance.java

```diff
@@ -72,6 +75,9 @@
      */
     private boolean synchronizationRegistered = false;
 
+    private final AtomicReference<Boolean> delayedAfterCompletion = new AtomicReference<>(false);
```

Comment:
There is an AtomicBoolean class that might fit better.  In any event though, the fact you have two Atomic* variables is a red flag.